src/lib/y2remote/remote.rb

```diff
@@ -0,0 +1,233 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LLC
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "y2remote/modes"
+
+module Y2Remote
+  class Remote
+    include Singleton
+    include Yast::Logger
+    include Yast::I18n
+
+    XDM_SERVICE_NAME = "display-manager".freeze
+    GRAPHICAL_TARGET = "graphical".freeze
+
+    PKG_CONTAINING_FW_SERVICES = "xorg-x11-Xvnc".freeze
+
+    # Currently, all attributes (enablement of remote access)
+    # are applied on vnc1 even vnchttpd1 configuration
+
+    # [Symbol] Remote administration mode, :disabled, :xvnc or :vncmanager
+    attr_reader :modes
+
+    attr_accessor :proposed
+
+    alias_method :proposed?, :proposed
+
+    def initialize
+      import_modules
+
+      textdomain "network"
+
+      @modes = []
+      @proposed = false
+    end
+
+    # Checks if remote administration is currently allowed
+    def enabled?
+      !disabled?
+    end
+
+    # Checks if remote administration is currently disallowed
+    def disabled?
+      modes.empty?
+    end
+
+    def disable!
+      @modes = []
+    end
+
+    # It add the given mode to the list of modes to be enabled
+    #
+    # @return [Array<Symbol>] list of enable vnc modes
+    def enable_mode(mode)
+      return modes if modes.include?(mode)
+
+      @modes.delete(:vnc) if mode == :manager
```

Comment:
np: I would prefer something like:

```ruby
if mode == :manager
  @modes.delete(:vnc)
elsif mode == :vnc
  @modes.delete(:manager)
end
```

Not a big deal, but it reads better IMHO.